webrender/src/glyph_rasterizer.rs

```diff
@@ -8,79 +8,68 @@ use api::{ColorF, ColorU};
 use api::{FontInstanceFlags, FontInstancePlatformOptions};
 use api::{FontKey, FontRenderMode, FontTemplate, FontVariation};
 use api::{GlyphDimensions, GlyphKey, LayoutToWorldTransform, SubpixelDirection};
-#[cfg(feature = "pathfinder")]
-use api::NativeFontHandle;
 #[cfg(any(test, feature = "pathfinder"))]
 use api::DeviceIntSize;
-#[cfg(not(feature = "pathfinder"))]
-use api::{ImageData, ImageDescriptor, ImageFormat};
 use app_units::Au;
-#[cfg(not(feature = "pathfinder"))]
-use device::TextureFilter;
-#[cfg(feature = "pathfinder")]
-use euclid::{TypedPoint2D, TypedSize2D, TypedVector2D};
 use glyph_cache::{CachedGlyphInfo, GlyphCache, GlyphCacheEntry};
 use gpu_cache::GpuCache;
-use gpu_types::UvRectKind;
 use internal_types::ResourceCacheError;
-#[cfg(feature = "pathfinder")]
-use pathfinder_font_renderer;
-#[cfg(feature = "pathfinder")]
-use pathfinder_partitioner::mesh::Mesh as PathfinderMesh;
-#[cfg(feature = "pathfinder")]
-use pathfinder_path_utils::cubic_to_quadratic::CubicToQuadraticTransformer;
 use platform::font::FontContext;
 use profiler::TextureCacheProfileCounters;
 use rayon::ThreadPool;
-#[cfg(not(feature = "pathfinder"))]
-use rayon::prelude::*;
 #[cfg(test)]
 use render_backend::FrameId;
 use render_task::{RenderTaskCache, RenderTaskTree};
-#[cfg(feature = "pathfinder")]
-use render_task::{RenderTask, RenderTaskCacheKey, RenderTaskCacheEntryHandle};
-#[cfg(feature = "pathfinder")]
-use render_task::{RenderTaskCacheKeyKind, RenderTaskId, RenderTaskLocation};
-#[cfg(feature = "pathfinder")]
-use resource_cache::CacheItem;
 use std::cmp;
 use std::collections::hash_map::Entry;
 use std::f32;
 use std::hash::{Hash, Hasher};
 use std::mem;
-#[cfg(feature = "pathfinder")]
-use std::ops::Deref;
 use std::sync::{Arc, Mutex, MutexGuard};
 use std::sync::mpsc::{channel, Receiver, Sender};
 use texture_cache::TextureCache;
-#[cfg(not(feature = "pathfinder"))]
-use texture_cache::TextureCacheHandle;
 #[cfg(test)]
 use thread_profiler::register_thread_with_profiler;
-#[cfg(feature = "pathfinder")]
-use tiling::RenderTargetKind;
 use tiling::SpecialRenderPasses;
-#[cfg(feature = "pathfinder")]
-use webrender_api::{DeviceIntPoint, DevicePixel};
+cfg_if! {
```

Comment:
The amount of diversion between the pathfinder and whatnot features of this module is pretty high. We should consider splitting those into separate modules, potentially leaving something shared in another module. This would be cleaner than multiple `#[cfg(feature = "pathfinder")]` spread over the code.

Reply:
Yeah ok, but this is going to take a while to split up the files and resolve all the errors. I'll do two seperate files for the pathfinder and the no_pathfinder configuration.

Reply:
btw, is there any difference between `use api::` and `use webrender_api::` or are they the same?

Reply:
No, this is because of this line in "lib.rs": `pub use webrender_api as api;`
I'd much prefer us just doing `pub extern crate webrender_api as api;` in the first place, without any ambiguity.